### ISSUE TITLE
[Question] Popups for other LSP features

### ISSUE THREAD

**Issue by tomaskallup** [RishabhRD/nvim-lsputils#22]
Hello and thank you for this great plugin!

I was wondering, if you had plans for other features, such as a popup for diagnostics (virtual text is horrible when it comes to multiline diagnostics) or maybe better hover? If so, I'll gladly create separate issues for the features I think would be usefull and help with some of them (the codebase seems simple enough for my level of expertise).

**Comment by RishabhRD** [comment]
Hello @tomaskallup I would be glad to add new features as user's requirement. I may be a little slow(hopefully in terms of a day or two) in implementation. Feel free to open new issues regarding any feature request. We can discuss on there. Your contribution would be so helpful for plugin!

**Comment by tomaskallup** [comment]
What I was thinking about these:
 -  rename popup, but I can see there is already an issues for that one
 -  diagnostics, having a nicer window with something like a title for the source of the diagnostic would be nice. An example would be [lps-saga's version](https://github.com/glepnir/lspsaga.nvim#jump-diagnostic-and-show-diagnostics) - I started tinkering with the diagnostics and will probably try to make that one happen.
 - hover, again, looking at [lsp-saga](https://github.com/glepnir/lspsaga.nvim#hover-doc) it has nice hover popup with scrolling (scrolling wouldn't be required, as you can always switch into the window and scroll)

I do however have a question. Why did you create your own implementation of popup? There already is [popup.nvim](https://github.com/nvim-lua/popup.nvim), is it missing some important features for this plugin?

**Comment by wucke13** [comment]
Any news on this? I'm also intrigued by having beautiful code actions and ugly diagnostics

**Comment by RishabhRD** [comment]
@wucke13 Yeah! I am doing some work on that but not on nvim-lsputils as I have made many development mistakes as I was during my initial phase of my undergrad. Is sometime I would deprecate the project.

Instead I am maintaining [lspactions](https://github.com/RishabhRD/lspactions) that is quite experimental and focus more on nvim-0.6 nightly. However it works fine for neovim 0.5.1 branch too. I would be open for new ideas based on lsp too for that project.

Regarding diagnostics, currently it doesn't have very pretty diagnostics. I focused more on other features. It would be part of lspactions soon. Currently it provides more things like renaming feature and yes rename buffer doesn't have any problems of classic neovim prompt buffer, and every module is way more robust and way more extensible. Like you can sort your codeactions based on any parameter you like.
